internal/services/domainservices/active_directory_domain_service_data_source.go

```diff
@@ -220,32 +220,42 @@ func dataSourceActiveDirectoryDomainServiceReplicaSetSchema() map[string]*plugin
 
 func dataSourceActiveDirectoryDomainServiceRead(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).DomainServices.DomainServicesClient
+	subscrptionId := meta.(*clients.Client).Account.SubscriptionId
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
 	name := d.Get("name").(string)
 	resourceGroup := d.Get("resource_group_name").(string)
 
-	resp, err := client.Get(ctx, resourceGroup, name)
+	idsdk := domainservices.NewDomainServiceID(subscrptionId, resourceGroup, name)
+
+	resp, err := client.Get(ctx, idsdk)
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if response.WasNotFound(resp.HttpResponse) {
 			return nil
 		}
 		return err
 	}
 
-	if resp.ID == nil {
+	model := resp.Model
+	if model == nil {
+		return fmt.Errorf("reading Domain Service: model was returned nil")
+	}
+
+	if model.Id == nil {
 		return fmt.Errorf("reading Domain Service: ID was returned nil")
 	}
-	d.SetId(*resp.ID)
+	d.SetId(*model.Id)
```

Comment:
is there a reason we're using this ID rather than the Resource ID above? they aren't guaranteed to be the same casing-wise?

Reply:
@tombuildsstuff I also noticed this, whilst I don't know whether changing this to the resource ID will introduce a breaking change. The intent of this PR is to switch the SDK verbatim.

Reply:
We should move also move towards using our generated IDs over the model ids in this swap too. Let's check to see if the ids are different too. If they are, we need to write a migration on top of it

Reply:
@mbfrahry I've made the change.